packages/salesforcedx-vscode-core/src/context/workspaceOrgType.ts

```diff
@@ -27,17 +27,22 @@ export async function getWorkspaceOrgType(
     throw e;
   }
 
+  const email = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(
```

Comment:
This regex fails if you have consecutive `-` characters or a combination of `.-`. e.g. test2342--sd@test.co

Reply:
POSIX expressions are not supported by javascript out of the box- you can download an npm package called `posix-character-classes` to use them. For now, I changed the regex to accept multiple special characters for the local part of the email.

Reply:
Thanks for checking, I think that just changing the regex for now should be enough